src/main/resources/avro/readmethods.avdl

```diff
@@ -14,8 +14,10 @@ to `ReferenceSet`s containing that same `Reference`. If no reference is
 specified, all `ReadGroup`s must be aligned to the same `ReferenceSet`.
 */
 record SearchReadsRequest {
-  /** If nonempty, restrict this query to reads within the given readgroups. */
-  array<string> readGroupIds = [];
+  /**
+  The ReadGroup to search.
+  */
+  string readGroupId;
```

Comment:
Hi David,

Welcome back :)  I like the points your brig up and I agree that we need to find a good balance, so maybe we should try to have that discussion now.  Let's take the Google Maps API, how much is performed on the server vs the client?  The data throughput for today's analysis would be of a similar scale.  Would one perform the [Google Distance Matrix API](https://developers.google.com/maps/documentation/distancematrix/) on their client?  Most likely not.  So we want to keep petabyte sub-second response times in mind as this API and it's implementations with distributed repositories scales.

Regarding adapting the API by taking into account currently implemented versions, has pros and cons.  On the pro-side it would tell us what is possible and what to tweak.  On the con-side, we do not want it to limit us.  I have dealt in the past with similar situations on very-large implementations, and one definitely does not want to migrate the data from one schema to another.  It is a huge pain, which I could image it to be at Google if the schema changes significantly here.  But let me ask you this.  What if in the future we brainstorm a major re-write that would improve the overall implementation?  Should we go forward, or should we stay away if the currently implemented versions of the API would break?  I think the goal is to work through these versions (since we are still sub-1.0 on GA4GH API) until we find the optimal setup.

What do you think?

Thanks,
Paul